src/dataflow/src/source/mod.rs

```diff
@@ -405,20 +406,28 @@ impl Codec for SourceTimestamp {
         buf.extend(&self.offset.offset.to_le_bytes());
     }
 
-    fn decode<'a>(buf: &'a [u8]) -> Result<Self, String> {
+    fn decode<'a>(buf: &'a [u8]) -> Result<Self, CodecError> {
         let typ = buf[0];
         let (partition, offset_start) = match typ {
             SOURCE_TIMESTAMP_PARTITION_KAFKA => {
                 let slice = &buf[1..(1 + 4)];
-                let pid =
-                    i32::from_le_bytes(<[u8; 4]>::try_from(slice).map_err(|err| err.to_string())?);
+                let pid = i32::from_le_bytes(
+                    <[u8; 4]>::try_from(slice).map_err(|err| CodecError::from(err.to_string()))?,
```

Comment:
Same here for avoiding `map_err` with a `impl From<std::array::TryFromSliceError> for CodecError` implementation